pkg/dataplane/server/server.go

```diff
@@ -0,0 +1,179 @@
+package server
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	cpapi "github.com/clusterlink-net/clusterlink/pkg/controlplane/api"
+	"github.com/clusterlink-net/clusterlink/pkg/dataplane/api"
+	"github.com/clusterlink-net/clusterlink/pkg/util/sniproxy"
+	"github.com/clusterlink-net/clusterlink/pkg/utils/netutils"
+)
+
+const (
+	httpSchemaPrefix = "https://"
+)
+
+// StartDataplaneServer starts the Dataplane server
+func (d *Dataplane) StartDataplaneServer(dataplaneServerAddress string) error {
+	d.logger.Infof("Dataplane server starting  at %s", dataplaneServerAddress)
+	server := netutils.CreateResilientHTTPServer(dataplaneServerAddress, d.router, d.parsedCertData.ServerConfig(), nil, nil, nil)
+
+	return server.ListenAndServeTLS("", "")
+}
+
+// StartSNIServer starts the SNI Proxy in the dataplane
+func (d *Dataplane) StartSNIServer(dataplaneServerAddress string) error {
+	dataplaneListenAddress := ":" + strconv.Itoa(api.ListenPort)
+	sniProxy := sniproxy.NewServer(map[string]string{
+		d.peerName:                          d.controlplaneTarget,
+		api.DataplaneServerName(d.peerName): dataplaneServerAddress,
+	})
+
+	d.logger.Infof("SNI proxy starting at %s.", dataplaneListenAddress)
+	err := sniProxy.Listen(dataplaneListenAddress)
+	if err != nil {
+		return fmt.Errorf("unable to create listener for server on %s: %v",
+			dataplaneListenAddress, err)
+	}
+	return sniProxy.Serve()
+}
+
+func (d *Dataplane) addAuthzHandlers() {
+	d.router.Post(cpapi.DataplaneIngressAuthorizationPath, d.dataplaneIngressAuthorize)
```

Comment:
Oh I see, I thought that change you had suggested was related to the refactoring. Let me change to "/" and verify, if it works and change.

Reply:
Verified and changed it to "/" instead of the specific path.

Reply:
> Verified and changed it to "/" instead of the specific path.

Actually, my verification was incorrect earlier. Upon revalidating. This change doesn't work. It needed the following commit as well : https://github.com/clusterlink-net/clusterlink/pull/58/commits/7b363c0cc9e6e69f38a59ec4ebf88a7370698f8a